lua/entities/gmod_wire_expression2/core/vector2.lua

```diff
@@ -1096,8 +1096,8 @@ __e2setcost(7)
 -- Returns a random vector4 between -1 and 1
 e2function vector4 randvec4()
 	local vec = { random()*2-1, random()*2-1, random()*2-1, random()*2-1 }
-	local length = sqrt( vec[1]^2+vec[2]^2+vec[3]^2+vec[4]^2 )
-	return { vec[1] / length, vec[2] / length }
+	local length = ( vec[1]^2+vec[2]^2+vec[3]^2+vec[4]^2 ) ^ 0.5 -- x ^ 0.5 <=> math.sqrt( x )
```

Comment:
You can expect `math.sqrt` to be faster than `x ^ 0.5` - they'll likely use different internal methods for converging on the correct answer, of which sqrt's would be more specialised for the special case of square roots. Moreover, it's a clarity issue - if you write `y = f(x) -- f(x) <=> g(x)`, and calling `g(x)` is no worse than `f(x)`, you should probably just call `g(x)` and drop the comment.